mongomock/filtering.py

```diff
@@ -508,6 +508,12 @@ def operator_eq(doc_val, search_val):
         isinstance(v, int) and not isinstance(v, bool) and v.bit_length() > 32
     ),
     'decimal': (lambda v: isinstance(v, Decimal128)) if Decimal128 else None,
+    'number': lambda v: any((
+        _type_op(v, 'int'),
```

Comment:
Please use a simpler `isinstance(v, (int, float) + ((Decimal128,) if Decimal 128 else ()))`

Reply:
Okay. Will do it. I'd like to notice though that I believe calling `_type_op()` here seems semantically more in tune with the meaning of the  definition of of type `number` by MongoDB. And less prone to errors too: for example, the simpler alternative proposed above is missing `and not isinstance(v, bool)`.

Reply:
Done.